modules/features/player/src/main/java/au/com/shiftyjelly/pocketcasts/player/view/transcripts/TranscriptViewModel.kt

```diff
@@ -1,65 +1,92 @@
 package au.com.shiftyjelly.pocketcasts.player.view.transcripts
 
 import androidx.annotation.OptIn
+import androidx.annotation.VisibleForTesting
 import androidx.lifecycle.ViewModel
 import androidx.lifecycle.viewModelScope
 import androidx.media3.common.Format
 import androidx.media3.common.util.UnstableApi
 import androidx.media3.extractor.text.CuesWithTiming
+import androidx.media3.extractor.text.CuesWithTimingSubtitle
 import androidx.media3.extractor.text.SubtitleParser
 import au.com.shiftyjelly.pocketcasts.models.entity.Podcast
 import au.com.shiftyjelly.pocketcasts.models.to.Transcript
+import au.com.shiftyjelly.pocketcasts.repositories.di.IoDispatcher
 import au.com.shiftyjelly.pocketcasts.repositories.playback.PlaybackManager
 import au.com.shiftyjelly.pocketcasts.repositories.podcast.TranscriptsManager
 import au.com.shiftyjelly.pocketcasts.utils.UrlUtil
 import com.google.common.collect.ImmutableList
 import dagger.hilt.android.lifecycle.HiltViewModel
+import java.util.regex.Pattern
 import javax.inject.Inject
+import kotlinx.coroutines.CoroutineDispatcher
 import kotlinx.coroutines.ExperimentalCoroutinesApi
-import kotlinx.coroutines.flow.SharingStarted
+import kotlinx.coroutines.flow.MutableStateFlow
 import kotlinx.coroutines.flow.StateFlow
 import kotlinx.coroutines.flow.distinctUntilChanged
 import kotlinx.coroutines.flow.flatMapLatest
 import kotlinx.coroutines.flow.map
 import kotlinx.coroutines.flow.stateIn
+import kotlinx.coroutines.launch
+import kotlinx.coroutines.withContext
 
+// TODO: [Transcript] Modify SpeakerRegex for non english languages
+private val SpeakerRegex = Pattern.compile("Speaker \\d*: ")
+private val NewLineRegex = Pattern.compile("\\.$")
```

Comment:
Should this also account for `?` and `!` characters?

And I think that similarly to the comment about speakers in non English languages this should also account for other languages like Japanese period `。`. This can be addressed in a separate PR but I wanted to note it.

Reply:
Good point. Included ? and ! characters also for new line regex in [eb83ed0](https://github.com/Automattic/pocket-casts-android/pull/2468/commits/eb83ed00789b93a32439868ae600bb274679691f)